config/config.go

```diff
@@ -349,16 +346,10 @@ func (c *Configuration) SetEndpointAccess(e Endpoint, authType string) {
 	switch authType {
 	case "", "none":
 		git.Config.UnsetLocalKey("", key)
-
-		c.loading.Lock()
-		delete(c.gitConfig, strings.ToLower(key))
-		c.loading.Unlock()
+		c.Git.Del(key)
```

Comment:
I definitely agree that this isn't my first choice of things to add to the public API for the `config` package, since it ~~encourages~~ allows calling code to preform these same mutable operations.

I think the `MutableEnvironment` is an interesting option. Here's another thought:

```go
func del(e *Environment, key string) *Environment { ... }
func set(e *Environment, key, val string) *Environment { ... }
```

and

```go
c.Git = del(c.Git, key)
```

This would allow us to make the mutable operations (`del` and `set`) package-private, while still supporting this single instance of mutation. I'm not immediately sure how you'd implement such a function "`del`", but on first glance it may be more complicated than simply making a copy of all keys save the one(s) that you want to remove.

Curious as to your thoughts!